Gate Serialization macros are not correctly exported

Hello guys!

I tried using macros from [gate_serialization.rs](https://github.com/0xPolygonZero/plonky2/blob/main/plonky2/src/util/serialization/gate_serialization.rs) to build custom gate serializer, but there were 2 problems:

```bash
error[E0433]: failed to resolve: use of undeclared crate or module `log`
  --> src/deserializer/custom.rs:28:5
   |
28 | /     impl_gate_serializer! {
29 | |         ZKVerifyGateSerializer,
30 | |         ArithmeticGate,
31 | |         ArithmeticExtensionGate<D>,
...  |
45 | |         ReducingGate<D>
46 | |     }
   | |_____^ use of undeclared crate or module `log`
   |
   = note: this error originates in the macro `get_gate_tag_impl` which comes from the expansion of the macro `impl_gate_serializer` (in Nightly builds, run with -Z macro-backtrace for more info)
```

```bash
error[E0603]: struct `Vec` is private
  --> /myuser/plonky2/plonky2/src/util/serialization/gate_serialization.rs:82:72
   |
82 |             buf: &mut $crate::util::serialization::gate_serialization::Vec<u8>,
   |                                                                        ^^^ private struct
   |
note: the struct `Vec` is defined here
  --> /myuser/plonky2/plonky2/src/util/serialization/gate_serialization.rs:6:5
   |
6  | use std::vec::Vec; // For macros below
   |     ^^^^^^^^^^^^^
help: consider importing one of these structs instead
   |
82 |             buf: &mut std::vec::Vec<u8>,
   |                       ~~~~~~~~~~~~~
82 |             buf: &mut plonky2::util::serialization::generator_serialization::Vec<u8>,
   |                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
help: import `Vec` directly
   |
82 |             buf: &mut alloc::vec::Vec,
   |                       ~~~~~~~~~~~~~~~

```

--- a/plonky2/src/util/serialization/gate_serialization.rs
+++ b/plonky2/src/util/serialization/gate_serialization.rs
@@ -6,7 +6,6 @@ pub use alloc::vec::Vec;
 pub use std::vec::Vec; // For macros below
 
 pub use log;
-
 use plonky2_field::extension::Extendable;
 
 use crate::gates::gate::GateRef;

--- a/plonky2/src/util/serialization/gate_serialization.rs
+++ b/plonky2/src/util/serialization/gate_serialization.rs
@@ -1,9 +1,11 @@
 //! A module to help with GateRef serialization
 
 #[cfg(not(feature = "std"))]
-use alloc::vec::Vec;
+pub use alloc::vec::Vec;
 #[cfg(feature = "std")]
-use std::vec::Vec; // For macros below
+pub use std::vec::Vec; // For macros below
+
+pub use log;
 
 use plonky2_field::extension::Extendable;
 
@@ -51,8 +53,8 @@ macro_rules! get_gate_tag_impl {
             Ok(tag)
         } else)*
         {
-            log::log!(
-                log::Level::Error,
+            $crate::util::serialization::gate_serialization::log::log!(
+                $crate::util::serialization::gate_serialization::log::Level::Error,
                 "attempted to serialize gate with id `{}` which is unsupported by this gate serializer",
                 $gate.0.id()
             );

